### PR TITLE
fix: support maps in fieldbehavior

### PR DESCRIPTION
**Pull request by ericwenn**

### CODE REVIEW

File: examples/proto/src/einride/example/freight/v1/shipment.proto [hunk 1]
```diff
@@ -65,6 +65,9 @@ message Shipment {
 
   // The line items of the shipment.
   repeated LineItem line_items = 11;
+
+  // Annotations of the shipment.
+  map<string,string> annotations = 12;
```
Comment (odsod) [inline examples/proto/src/einride/example/freight/v1/shipment.proto:70]: Good idea!

File: fieldbehavior/fieldbehavior.go [hunk 2]
```diff
@@ -57,14 +57,20 @@ func CopyFields(dst, src proto.Message, behaviorsToCopy ...annotations.FieldBeha
...
+}
+
+func isPresent(v protoreflect.Value, f protoreflect.FieldDescriptor) bool {
 	if !v.IsValid() {
 		return false
 	}
 	if f.IsList() {
 		return v.List().Len() > 0
 	}
+	if f.IsMap() {
```
Comment (odsod) [inline fieldbehavior/fieldbehavior.go:71]: nice

File: fieldbehavior/required.go [hunk 3]
```diff
@@ -27,21 +27,39 @@ func validateRequiredFields(reflectMessage protoreflect.Message, mask *fieldmask
...
+			case field.IsList():
 				for i := 0; i < value.List().Len(); i++ {
 					if err := validateRequiredFields(value.List().Get(i).Message(), mask, currPath); err != nil {
 						return err
 					}
 				}
-			} else if err := validateRequiredFields(value.Message(), mask, currPath); err != nil {
-				return err
+			case field.IsMap():
+				if field.MapValue().Kind() == protoreflect.MessageKind {
```
Comment (odsod) [inline fieldbehavior/required.go:45]: might be easier to read if we switch the if around to do an early return/continue here?

**Review by odsod — APPROVED** [review]
Looks great! One suggestion on simplifying the control flow

File: fieldbehavior/required.go (hunk 3 above)
Comment (ericwenn) [reply fieldbehavior/required.go:45]: Agree. I think we can simplify this entire function a bit.
I'll change the if-order in this PR and will follow up with another change.

**Pull request merged**